cmake/EthDependencies.cmake

```
@@ -62,3 +62,27 @@ foreach (BOOST_COMPONENT IN LISTS BOOST_COMPONENTS)
 	get_property(LOCATION TARGET Boost::${BOOST_COMPONENT} PROPERTY IMPORTED_LOCATION)
 	message(STATUS "Found Boost::${BOOST_COMPONENT} at ${LOCATION}")
 endforeach()
+
+if(NOT USE_SYSTEM_LIBRARIES)
+	file(GLOB fmtlib_contents "${CMAKE_SOURCE_DIR}/deps/fmtlib/*")
+	file(GLOB nlohmannjson_contents "${CMAKE_SOURCE_DIR}/deps/nlohmann-json/*")
+	file(GLOB rangev3_contents "${CMAKE_SOURCE_DIR}/deps/range-v3/*")
+	if (fmtlib_contents AND nlohmannjson_contents AND rangev3_contents)
+		message(STATUS "git submodules seem to be already initialized: nothing to do.")
+	else()
+		message(STATUS "git submodules seem not to be initialized: implicitly executing 'git submodule update --init'.")
+		find_package(Git)
+		if (Git_FOUND)
+			execute_process(
+					COMMAND git submodule update --init
+					WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+					RESULT_VARIABLE result
+			)
+			if(NOT result EQUAL 0)
+				message(FATAL_ERROR "Failed to initialize submodules: 'git submodule update --init' failed.")
+			endif()
+		else()
+			message(FATAL_ERROR "Failed to initialize submodules: error executing 'git submodule update --init': git was not found.")
```

I mean, if this happens, you didn't really run the command so it's irrelevant. Only the fact that git is not installed matters:
```suggestion
			message(FATAL_ERROR "Failed to initialize submodules: 'git' command not found.")
```

---

Also, I'd generally recommend dealing with special cases first since it requires less nesting and makes flow clearer:
```cmake
if (NOT Git_FOUND)
    message(FATAL_ERROR ...)
endif()

execute_process(...)
...
```